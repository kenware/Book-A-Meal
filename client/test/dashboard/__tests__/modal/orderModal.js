import React from 'react';
import { shallow } from 'enzyme';
import OrderModal from '../../../../components/dashboard/modals/orderModal';
import { props, emptyProps } from '../../dashboardMockData';

describe('Test menu component of dashboard', () => {
  it('renders correctly with empty props', () => {
    const tree = shallow(<OrderModal {...emptyProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render with props', () => {
    const tree = shallow(<OrderModal {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render with props', () => {
    const tree = shallow(<OrderModal {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
