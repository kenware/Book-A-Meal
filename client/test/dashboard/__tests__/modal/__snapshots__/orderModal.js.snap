// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test menu component of dashboard renders correctly with empty props 1`] = `
<div
  className=""
>
  <br />
  <div
    className="modal-header"
  >
    <p
      className="justify l-r-pad-text"
    >
      This order can only be modified within ONE hour after it is created
      <h4
        className="danger text-center"
      />
    </p>
  </div>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text"
    >
       Name: 
    </span>
    <span
      className="modal-items "
    />
  </div>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text"
    >
       Price: 
    </span>
    <span
      className="modal-items "
    />
  </div>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text"
    >
       Quantity: 
    </span>
    <input
      className="modal-menu select"
      id="quantity"
      name="quantity"
      onChange={[MockFunction]}
      type="number"
    />
  </div>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text address"
    >
       Address: 
    </span>
    <textarea
      className="modal-menu"
      id="address"
      name="address"
      onChange={[MockFunction]}
      rows="4"
      type="text"
      value=""
    />
  </div>
  <button
    className="checkout-btn"
    onClick={[MockFunction]}
  />
</div>
`;

exports[`Test menu component of dashboard should render with props 1`] = `
<div
  className=""
>
  <br />
  <div
    className="modal-header"
  >
    <p
      className="justify l-r-pad-text"
    >
      This order can only be modified within ONE hour after it is created
      <h4
        className="danger text-center"
      />
    </p>
  </div>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text"
    >
       Name: 
    </span>
    <span
      className="modal-items "
    />
  </div>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text"
    >
       Price: 
    </span>
    <span
      className="modal-items "
    />
  </div>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text"
    >
       Quantity: 
    </span>
    <input
      className="modal-menu select"
      id="quantity"
      name="quantity"
      onChange={[MockFunction]}
      type="number"
    />
  </div>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text address"
    >
       Address: 
    </span>
    <textarea
      className="modal-menu"
      id="address"
      name="address"
      onChange={[MockFunction]}
      rows="4"
      type="text"
      value="Reveend street"
    />
  </div>
  <button
    className="checkout-btn"
    onClick={[MockFunction]}
  />
</div>
`;
